pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/timeseries/TimeSeriesDispatchObserver.java

```diff
@@ -30,37 +35,57 @@
  *   engine integration.
  */
 public class TimeSeriesDispatchObserver implements StreamObserver<Worker.TimeSeriesResponse> {
```

Comment:
Now as Broker is doing reduce work ? Can broker become bottleneck?  Did you perform any analysis for broker in your cluster testing?